File: src/api/trade/venues.ts
import type { HttpClient } from '../../utils/http.js';
import type { Venue } from '../../types/index.js';

/**
 * API for retrieving trading venue information.
 *
 * Venues represent the underlying exchanges or liquidity sources
 * where trades are executed.
 *
 * @example
 * ```typescript
 * const dflow = new DFlowClient();
 *
 * const venues = await dflow.venues.getVenues();
 * venues.forEach(venue => console.log(venue.name));
 * ```
 */
export class VenuesAPI {
  constructor(private http: HttpClient) {}

  /**
   * Get all available trading venues.
   *
   * @returns Array of venue information
   *
   * @example
   * ```typescript
   * const venues = await dflow.venues.getVenues();
   * venues.forEach(venue => {
   *   console.log(`${venue.name}: ${venue.description}`);
   * });
   * ```
   */
  async getVenues(): Promise<Venue[]> {
    return this.http.get<Venue[]>('/venues');
  }
}

File: python/src/dflow/api/trade/venues.py
"""Venues API for DFlow SDK."""

from dflow.types import Venue
from dflow.utils.http import HttpClient


class VenuesAPI:
    """API for retrieving trading venue information.

    Venues represent the underlying exchanges or liquidity sources
    where trades are executed.

    Example:
        >>> dflow = DFlowClient()
        >>> venues = dflow.venues.get_venues()
        >>> for venue in venues:
        ...     print(f"{venue.name}: {venue.label}")
    """

    def __init__(self, http: HttpClient):
        self._http = http

    def get_venues(self) -> list[Venue]:
        """Get all available trading venues.

        Returns:
            Array of venue information

        Example:
            >>> venues = dflow.venues.get_venues()
            >>> for venue in venues:
            ...     print(f"{venue.name}: {venue.label}")
        """
        data = self._http.get("/venues")
        return [Venue.model_validate(v) for v in data]
